Google.Api.Generator.Tests/ProtoTests/ChildResource/Testing.ChildResource/ChildResourceResourceNames.g.cs

```diff
@@ -22,67 +22,178 @@ namespace Testing.ChildResource
     /// <summary>Resource name for the <c>Project</c> resource.</summary>
     public sealed partial class ProjectName : gax::IResourceName, sys::IEquatable<ProjectName>
     {
-        private static readonly gax::PathTemplate s_template = new gax::PathTemplate("projects/{project}");
+        /// <summary>The possible contents of <see cref="ProjectName"/>.</summary>
+        public enum ResourceType
+        {
+            /// <summary>A resource of an unknown type.</summary>
+            Unknown = 0,
+
+            /// <summary>A resource of type 'project'.</summary>
+            Project = 1
+        }
+
+        private static gax::PathTemplate s_project = new gax::PathTemplate("projects/{project}");
+
+        /// <summary>Creates a <see cref="ProjectName"/> containing an unknown resource name.</summary>
+        /// <param name="unknownResourceName">The unknown resource name. Must not be <c>null</c>.</param>
+        /// <returns>
+        /// A new instance of <see cref="ProjectName"/> containing the provided <paramref name="unknownResourceName"/>.
+        /// </returns>
+        public static ProjectName CreateUnknown(gax::UnknownResourceName unknownResourceName) =>
+            new ProjectName(ResourceType.Unknown, gax::GaxPreconditions.CheckNotNull(unknownResourceName, nameof(unknownResourceName)));
+
+        /// <summary>Creates a <see cref="ProjectName"/> with the pattern <c>projects/{project}</c>.</summary>
+        /// <param name="projectId">The <c>Project</c> ID. Must not be <c>null</c> or empty.</param>
+        /// <returns>A new instance of <see cref="ProjectName"/> constructed from the provided ids.</returns>
+        public static ProjectName CreateProject(string projectId) =>
```

Comment:
I wonder whether we need a "Name" suffix on all of these, or at least something slightly different. Calling `CreateProject` sounds like it's going to create a project, for example. I know the class name suggests otherwise. Possibly `ForProject` or `FromProject` would be better?

Reply:
Renamed as `From...`